### PR TITLE
TP regularization and cross-validation

### PR DESCRIPTION
**Pull request by GTorlai**
This PR adds the TP regularization for process tomography, `runcircuit` with a single gate, cross-valildation on held-out data,  and it updates the `observer`.

### CODE REVIEW

File: examples/7_qpt_circuit.jl [hunk 1]
```diff
@@ -38,6 +38,7 @@ U = tomography(data, U0;
                optimizer = SGD(η = 0.1),
                batchsize = 500,
                epochs = 5,
+               tp_regularizer = 0.1,
```
Comment (mtfishman) [inline examples/7_qpt_circuit.jl:41]: Would be better to explicitly say `trace_preserving_regularizer`.

File: src/circuits/runcircuit.jl [hunk 2]
```diff
@@ -163,9 +163,11 @@ Apply a quantum gate to an MPS/MPO.
 function applygate!(M::Union{MPS,MPO},
                     gatename::String,
                     sites::Union{Int,Tuple};
+                    cutoff=1E-15,
+                    maxdim=1_000,
```
Comment (mtfishman) [inline src/circuits/runcircuit.jl:167]: Again, not a fan of putting an explicit maximum dimension. Also a reminder that we want to remove `applygate` in favor of `runcircuit`.

File: src/lpdo.jl [hunk 3]
```diff
@@ -83,7 +83,7 @@ function normalize!(M::MPO;
   return M
 end
 
-function normalize!(L::LPDO; sqrt_localnorms! = [])
+function normalize!(L::LPDO; sqrt_localnorms! = [], choi::Bool=false)
```
Comment (mtfishman) [inline src/lpdo.jl:86]: I would also prefer a more generic solution here, like generally being able to specify what norm you want to normalize to. For example, we could have a flag `localnorm` which defaults to 1, and then for a Choi matrix or unitary MPO, you can set `localnorm = 2`.

File: src/circuits/runcircuit.jl (hunk 2 above)
Comment (GTorlai) [reply src/circuits/runcircuit.jl:167]: I still need to fix a few things around, including this.

File: examples/7_qpt_circuit.jl (hunk 1 above)
Comment (GTorlai) [reply examples/7_qpt_circuit.jl:41]: OK

File: src/lpdo.jl (hunk 3 above)
Comment (GTorlai) [reply src/lpdo.jl:86]: Yes, I can add that

File: src/circuits/runcircuit.jl (hunk 2 above)
Comment (GTorlai) [reply src/circuits/runcircuit.jl:167]: I will also add the `runcircuit` with a single gate

**Pull request merged**